app/components/newsletter/newsletter-header-overview.js

```diff
@@ -77,25 +79,32 @@ export default class NewsletterHeaderOverviewComponent extends Component {
   @task
   *publishToMail() {
     try {
-      yield this.ensureMailCampaign();
-
-      if (this.mailCampaign?.isSent) {
-        this.toaster.error(this.intl.t('error-already-sent-newsletter'));
-      } else {
-        if (yield this.validateMailCampaign()) {
-          try {
-            yield this.newsletterService.sendMailCampaign(this.mailCampaign.campaignId);
-            this.mailCampaign.sentAt = new Date();
-            yield this.mailCampaign.save();
-            this.toaster.success(this.intl.t('success-publish-newsletter-to-mail'));
-          } catch(e) {
-            console.log("error sending newsletter", e);
-            this.toaster.error(
-              this.intl.t('error-send-newsletter'),
-              this.intl.t('warning-title')
-            );
+      if (yield this.canSendMailCampaign()) {
+        yield this.ensureMailCampaign();
+
+        if (this.mailCampaign?.isSent) {
```

Comment:
I know this was already here before but `?` is a bit silly right after doing `this.ensureMailCampaign()`